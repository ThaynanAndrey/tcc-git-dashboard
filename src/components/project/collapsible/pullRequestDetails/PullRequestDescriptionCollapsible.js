import React from 'react';
import showdown from 'showdown';
import CollapsibleCard from '../../../layout/CollapsibleCard';

const styleCollapsibleHeader = {
    display: "flex",
    justifyContent: "space-between"
};

const styleCollapsibleHeaderText = {
    display: "flex"
};

/**
 * Stateless component with the function of presenting a Collapsible to Pull Requests.
 * 
 * @author Thaynan Nunes
 */
const PullRequestDescriptionCollapsible = ({ pullRequest }) => {
    
    /**
     * Gets the html desciption from Markdown text.
     */
    const getHtmlMarkdownText = (textMD) => {
        if(textMD && document.getElementById("markdown-description")) {
            const converter = new showdown.Converter();
            const html = converter.makeHtml(textMD);
            document.getElementById("markdown-description").innerHTML = html;
        } else if(document.getElementById("markdown-description")) {
            document.getElementById("markdown-description").innerHTML = "<div>Não há descrição para esse Pull Request!</div>"
        }
    }

    return (
        <CollapsibleCard isActive={false}>
            <div className="collapsible-header" style={styleCollapsibleHeader}>
                <div style={styleCollapsibleHeaderText}>
                    <i className="material-icons">dehaze</i>
                    Descrição
                </div>
            </div>
            <div id="markdown-description"
                    className="collapsible-body body-collapsible-project descriptionPullRequestContent">
                {getHtmlMarkdownText(pullRequest.description)}
            </div>
        </CollapsibleCard>
    );
};

export default PullRequestDescriptionCollapsible;