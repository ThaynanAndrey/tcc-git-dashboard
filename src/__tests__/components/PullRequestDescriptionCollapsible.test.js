import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestDescriptionCollapsible from '../../components/project/collapsible/pullRequestDetails/PullRequestDescriptionCollapsible';

describe('Test PullRequestDescriptionCollapsible', () => {
  let tree;
  
  beforeEach(() => {
    const pullRequest = {};

    tree = shallow(
      <PullRequestDescriptionCollapsible
        pullRequest={pullRequest}
      />
    );
  });

  it('renders correctly a component to show PullRequestDescriptionCollapsible', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  
  it('the component should have class collapsible-header', () => {
    expect(tree.find(".collapsible-header").exists()).toBeTruthy();
  });

  it('the component should have classes collapsible-body, body-collapsible-project and descriptionPullRequestContent', () => {
    const pullRequest = {
        description: "###Apenas um teste!"
    };

    tree = shallow(
      <PullRequestDescriptionCollapsible
        pullRequest={pullRequest}
      />
    );

    expect(tree.find(".collapsible-body.body-collapsible-project.descriptionPullRequestContent").exists()).toBeTruthy();
  });
})