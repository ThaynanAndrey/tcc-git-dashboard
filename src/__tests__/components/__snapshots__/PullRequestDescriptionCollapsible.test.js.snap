// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestDescriptionCollapsible renders correctly a component to show PullRequestDescriptionCollapsible 1`] = `
<CollapsibleCard
  isActive={false}
>
  <div
    className="collapsible-header"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <i
        className="material-icons"
      >
        dehaze
      </i>
      Descrição
    </div>
  </div>
  <div
    className="collapsible-body body-collapsible-project descriptionPullRequestContent"
    id="markdown-description"
  />
</CollapsibleCard>
`;
